tests/test_config_entries.py

```diff
@@ -2299,6 +2303,72 @@ async def async_step_import(self, user_input):
         assert entries[0].state is config_entries.ConfigEntryState.LOADED
 
 
+async def test_async_setup_init_entry_completes_before_loaded_event_fires(hass):
```

Comment:
I saw reports of the sun entity not being present so I added this as I wanted to validate the order of events given there is a comment in `setup.py` about being fragile https://github.com/home-assistant/core/blob/dev/homeassistant/setup.py#L264